src/main/java/coresearch/cvurl/io/request/RequestBuilder.java

```diff
@@ -102,13 +106,21 @@ public T timeout(Duration timeout) {
         return (T) this;
     }
 
+    @SuppressWarnings("unchecked")
+    public T compressedResponse() {
+        this.compressedResponse = true;
+        this.header(HttpHeader.ACCEPT_ENCODING, HttpContentEncoding.GZIP);
```

Comment:
I think will be better to set header in setUpHttpRequestBuilder() method

Reply:
why ?

Reply:
to prevent the possibility to set an incorrect header after calling compressedResponse() method.
e.g.

var response = cvurl.POST(url)
                .compressedResponse()
                .header(HttpHeader.ACCEPT_ENCODING, "invalid_value")
                .build().asStream().orElseThrow(RuntimeException::new);

Reply:
yeah good point, will do